lock/common_test.go

```diff
@@ -0,0 +1,293 @@
+package lock
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type LockSuite struct {
+	suite.Suite
+	NewService func() Service
+}
+
+func (s *LockSuite) TestLockUnlockSingleLockNoConcurrency() {
+	assert := s.Assert()
+	niter := 100
+	id := "mylock"
+	service := s.NewService()
+	cfg := &SessionConfig{TTL: 1 * time.Second}
+	session, err := service.NewSession(cfg)
+	assert.NoError(err)
```

Comment:
instead of call to `assert.Blabla()`, you can call directly to `s.Blabla()` for assertions. Maybe instead of use the `assert` package, we want to use `require` one.